polyfill/intersection-observer.js

```diff
@@ -131,6 +131,12 @@ IntersectionObserver.prototype.THROTTLE_TIMEOUT = 100;
  */
 IntersectionObserver.prototype.POLL_INTERVAL = null;
 
+/**
+ * Use a mutation observer on the root element 
+ * to detect intersection changes.
+ */
+IntersectionObserver.prototype.USE_MUTATION_OBSERVER = true;
```

Comment:
I don't think this is part of the spec, so shouldn't be here. Otherwise, you should use a ES6 Symbol, right?

Reply:
Also, it's unclear if user land code should be able to set this? Should it be read only.

Reply:
I'm not using Symbol because this is a polyfill and should be compatible with a wide range of user agent. Also this is the way this polyfill already expose flags https://github.com/w3c/IntersectionObserver/blob/master/polyfill/intersection-observer.js#L117 

Also this PR addresses #261 which @philipwalton seems to be ok with

Reply:
@marcoscaceres the mutationobserver makes the polyfill unusable on certain browsers and consumes more resources than are often necessary so it definitely should be configurable by users.

I agree with @JSteunou that this is exactly like the other flags and is implemented correctly.

Reply:
> @marcoscaceres the mutationobserver makes the polyfill unusable on certain browsers and consumes more resources than are often necessary so it definitely should be configurable by users.

I'm completely sympathetic to what you are saying @bpossolo, but if there are clear perf wins from having when not using mutation observers, then always take the middle ground. If the polyfill starts sprouting additional attributes, etc. then it's no longer a polyfill, as it no longer matches the spec, which can come back to bite us:

https://www.w3.org/2001/tag/doc/polyfills/

Cc @bkardell

Reply:
fwiw, I'm relaxed about polyfills exposing non-spec attributes if their implementation incorporates trade offs that developers might want access to / have an opinion on.  It's also sometimes useful to make the polyfill detectable.  So the principle of perfect spec conformance is not hard and fast for me (or for the TAG polyfill recommendations).  In this specific case, I think I suggested using MOs in the first place, to avoid polling.  Polyfill.io tries to avoid shipping polyfills that poll, because they make it hard to debug applications when your flamechart is basically just noise.

Reply:
The way I view it...the spec is simply an interface and the implementations of any interface will naturally differ and should be configurable.

This implementation chooses to use a MutationObserver and polling.
The native browser implementation likely uses something completely different and will likely expose their own native properties on their objects.